src/main/java/ch/njol/skript/expressions/ExprDurability.java

```diff
@@ -66,53 +70,73 @@ public String getPropertyName() {
 	}
 	
 	@Override
-	public Class<Short> getReturnType() {
-		return Short.class;
+	public Class<Object> getReturnType() {
+		return Object.class;
 	}
 	
 	@Override
 	@Nullable
 	public Class<?>[] acceptChange(final ChangeMode mode) {
-		if (mode == ChangeMode.REMOVE_ALL)
-			return null;
-		if (Slot.class.isAssignableFrom(getExpr().getReturnType()) || getExpr().isSingle() && ChangerUtils.acceptsChange(getExpr(), ChangeMode.SET, ItemStack.class, ItemType.class))
-			return CollectionUtils.array(Number.class);
-		return null;
+		return (mode == ChangeMode.REMOVE_ALL) ? null : CollectionUtils.array(Object.class);
 	}
 	
+	@SuppressWarnings("deprecation")
 	@Override
 	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) {
```

Comment:
i feel like this whole method is really confusingly written (particularly how the recursion is involved)